### PR TITLE
Swap out autotools for meson build system

### LINKED ISSUE #44: Use meson as the build system
We currently use autotools and there is an experimental cmake branch, we need to come to a decision about which way we're going on this so we can start integrating it more tightly.

### PR DESCRIPTION
**Pull request by foolswood**
Resolves #44.

Looking at this I am wondering whether we should call it libadiff as that's a more canonical name, but that's a separate thing.

### CODE REVIEW

**Comment by ch3pjw** [comment]
Raised #49 for the name.

File: Notes [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # To build:
-autoreconf -i
-CC="gcc-4.8.3" CFLAGS="-fms-extensions -std=c11" ./configure && make
-# You shouldn't need that exact version of gcc, it just has to be recent and
-# you shouldn't really need the -fms-extensions either if yours is up to date
-# enough.
+mkdir build
```
Comment (ch3pjw) [inline Notes]: You could just whap a `.gitignore` in a pre-made directory to stop everyone having to do this. See [this popular StackOverflow answer](https://stackoverflow.com/questions/115983/how-can-i-add-an-empty-directory-to-a-git-repository)

File: Notes [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+meson . build
```
Comment (ch3pjw) [inline Notes]: I just did `meson build` according to the instructions on http://mesonbuild.com/Quick-guide.html when playing with [rage](https://github.com/concert/rage). Which is preferred and why?

Comment (foolswood) [reply Notes]: That'd be fine, I think what I was doing might be an old way.

File: Notes (hunk 1 above)
Comment (foolswood) [reply Notes]: That's what I did for rage, and it was sort of good, and sort of really annoying.
When you want to clear your build directory you can't just delete it with impunity due to the gitignore.

**Pull request merged**